src/api/store/supported_language.py

```diff
@@ -143,10 +144,15 @@ def list_campaign_supported_languages(self, context, args):
             if not campaign_id:
                 return None, CustomMassenergizeError("Please provide a campaign_id")
             
-            campaign = Campaign.objects.filter(pk=campaign_id).first()
+            campaign = None
+            try:
+                uuid_id = UUID(campaign_id, version=4)
+                campaign = Campaign.objects.filter(id=uuid_id, is_deleted=False).first()
+            except ValueError:
+                campaign = Campaign.objects.filter(slug=campaign_id, is_deleted=False).first()
             
             if not campaign:
-                return None, CustomMassenergizeError("Campaign not found")
+                return None, CustomMassenergizeError("Campaign with id does not exist")
```

Comment:
Can you externalise the error message to make it reusable and also show the actual id that wasn't found?

Reply:
updated